e2etest/bgptests/setup.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"github.com/onsi/ginkgo"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	"go.universe.tf/metallb/e2etest/pkg/config"
+	frrcontainer "go.universe.tf/metallb/e2etest/pkg/frr/container"
+	"go.universe.tf/metallb/e2etest/pkg/metallb"
+	testservice "go.universe.tf/metallb/e2etest/pkg/service"
+	metallbconfig "go.universe.tf/metallb/internal/config"
+	"go.universe.tf/metallb/internal/ipfamily"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eservice "k8s.io/kubernetes/test/e2e/framework/service"
+)
+
+func setupBGPService(f *framework.Framework, pairingIPFamily ipfamily.Family, poolAddresses []string, tweak testservice.Tweak) (*e2eservice.TestJig, *corev1.Service) {
```

Comment:
should we use this function in other tests as well?

Reply:
I wasn't sure. These tests are the "plain" ones, the others are (almost all) slight variations, and I did not want to tamper this PR with too much content.